### PR TITLE
Chao/xccl ut

### PR DESCRIPTION
**Pull request by Chao1Han**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def register_backend(
         Backend.backend_list.append(name.lower())
         if devices is not None:
             for device in devices:
-                if device != "cpu" and device != "cuda":
+                if device != "cpu" and device != "cuda" and device != "xpu":
```
Comment (zhangxiaoli73) [inline torch/distributed/distributed_c10d.py:346]: Do we really need to change it?

File: torch/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -780,6 +780,7 @@ def forward(self, input: Tensor) -> Tensor:
             # currently only GPU/PrivateUse1 input is supported
             if input.device.type not in [
                 "cuda",
+                "xpu",
```
Comment (zhangxiaoli73) [inline torch/nn/modules/batchnorm.py:783]: Changes done in latest stock PyTorch

File: torch/testing/_internal/distributed/distributed_test.py [hunk 3]
```diff
@@ -435,6 +435,8 @@ def check(backend):
             return dist.is_mpi_available()
         if backend == dist.Backend.UCC:
             return dist.is_ucc_available()
+        if backend == dist.Backend.CCL:
+            return dist.is_ccl_available()
```
Comment (zhangxiaoli73) [inline torch/testing/_internal/distributed/distributed_test.py:439]: `dist.Backend.XCCL`
